site/docs/spec/terminology.yaml

```diff
@@ -0,0 +1,230 @@
+  the fields map to the column {data type}s.
+relation: >-
+  an operator that converts zero or more input {dataset}s to a new {dataset}.
+
+static: >-
+  something that is defined before a {plan} is executed, and doesn't change
+  during its execution. Contrasts with {runtime}.
+runtime: >-
+  something that is computed while a {plan} is executed. Contrasts with {static}.
+
+predefined: >-
+  qualifier for something defined as part of this specification. Contrasts with
+  {user-defined}.
+user-defined: >-
+  qualifier for something defined within an {extension}. Contrasts with
+  {predefined}.
+
+extension: >-
+  something defined outside the context of this specification that may be used
+  in a plan by means of a URI-based link to a {simple extension} or a protobuf
+  `Any` message (a.k.a. {advanced extension}).
+simple extension: >-
+  an {extension} to Substrait, normally defined by means of a YAML file
+  compliant with the
+  [simple extension schema](https://github.com/substrait-io/substrait/blob/main/text/simple_extensions_schema.yaml),
+  referred to from a plan by means of an URI. Simple extensions can currently
+  define {function}s, {type class}es, and {type variation}s.
+advanced extension: >-
+  an {extension} to Substrait, used within a plan by populating one of the
+  various protobuf `Any` messages contained therein.
+
+anchor: >-
+  an integer in the protobuf serialization that can be referred to by
+  means of {reference}s elsewhere in a {plan}. Anchors are used for
+  {simple extension} URIs, {user-defined} {function}s, {user-defined}
+  {type class}es, and {user-defined} {type variation}s. Anchors are namespaced
+  by their type, but must otherwise be unique within the plan. Their value has
+  no significance other than to be a unique thing to refer to.
+reference: >-
+  an integer in the protobuf serialization that can be used to refer to
+  an {anchor}.
+
+function: >-
+  some {user-defined} operation on some information (usually specified by means
+  of {function argument}s) that yields a single {datum} for each invocation.
+  Functions are declared via {function declaration}s in {simple extension}s,
+  and referred to from within a plan via {function binding}s. They are
+  subdivided into {scalar function}s, {aggregate function}s, and
+  {window function}s.
+scalar function: >-
+  a {function} that computes a {datum} for each row of a {dataset} individually
+  within {expression} context, as defined
+  [here](../expressions/scalar_functions.md).
+aggregate function: >-
+  a {function} that computes a single {datum} for a complete {dataset}, as
+  defined [here](../expressions/aggregate_functions.md).
+window function: >-
+  a {function} that computes a {datum} for each row of a {dataset}, making use
+  of some window of surrounding rows in addition to the row it is computed for,
+  as defined [here](../expressions/window_functions.md).
+
+function declaration: >-
```

Comment:
I've been calling the whole function function declaration and the specific implementation function signature. It seems like you're missing the latter concept?